packages/editor/src/store/private-actions.js

```diff
@@ -241,7 +243,12 @@ export const revertTemplate =
 	async ( { registry } ) => {
 		const noticeId = 'edit-site-template-reverted';
 		registry.dispatch( noticesStore ).removeNotice( noticeId );
-		if ( ! isTemplateRevertable( template ) ) {
+
+		const isRevertable =
+			template.type === TEMPLATE_PART_POST_TYPE
+				? isTemplatePartRevertable
+				: isTemplateRevertable;
+		if ( ! isRevertable( template ) ) {
 			registry
 				.dispatch( noticesStore )
 				.createErrorNotice( __( 'This template is not revertable.' ), {
```

Comment:
The message needs to reflect the `type`, but the fix needs to be backported into RC, and we're currently in the "hard string-freeze" phase.

I'll handle string updates separately.